RFC0003.md

```diff
@@ -0,0 +1,179 @@
+
+Accordingly, this RFC proposes to modify the current PMIx\_Connect[\_nb], PMIx\_Disconnect[\_nb], and their associated server-RM interfaces to accommodate the requests. It is recognized that this causes a backward-compatibility issue. However, only one library currently uses these interfaces (and they are accessed via an abstraction layer that can hide the changes) - thus, the disruption should be minimal provided these changes can be adopted and implemented before other libraries adopt them.
+
+Some additional info keys are provided for the revised operations:
+
+* PMIX\_CONNECT\_NOTIFY\_EACH: generate a local event notification using the PMIX\_PROC\_HAS\_CONNECTED event each time a process connects, and the PMIX\_PROC\_HAS\_DISCONNECTED event each time a process disconnects. The PMIx server is responsible for generating these events.
+
+* PMIX\_CONNECT\_NOTIFY\_REQ: notify each of the indicated procs that they are requested to connect using the PMIX\_CONNECT\_REQUESTED event. The PMIx server is responsible for generating this event.
+
+* PMIX\_CONNECT\_OPTIONAL: participation is optional - do not return error if procs terminate without having connected. This is the joint responsibility of the host RM (to notify if remote procs terminate early) and the PMIx server.
+
+* PMIX\_CONNECT\_EARLY\_TERM\_ALLOWED: do not generate an error if a process terminates without disconnecting. The host RM and PMIx server are responsible for adjusting completion requirements on any ongoing or future collectives for this nspace.
+
+* PMIX\_CONNECT\_XCHG\_ONLY: ensure that all participating processes have a copy of the job-level info from each nspace, but do not assign a new nspace and rank
+
+
+#### PMIx\_Connect
+Record the specified processes as "connected". Both blocking and non-blocking
+versions are provided. This means that the resource manager should treat the
+failure of any process in the specified group as a reportable event, and take
+appropriate action. Note that different resource managers may respond to
+failures in different manners.
+
+The host RM will assign a unique nspace to the resulting process group. The
+new nspace must be provided when disconnecting from the group. All procs are
+required to disconnect from the nspace prior to terminating.
+
+Note: any single process _can_ be simultaneously engaged
+in multiple connect operations. For scalability, PMIx does not use a
+collective to assign an identifier to the connect operation. Instead, it
+is expected that the provided array of process IDs will serve to identify
+a specific PMIx\_Connect operation. This means, however, that there can
```

Comment:
Does the order of the array distinguish it or is it the set of entries that distinguish it?

Example (nspace=1, {NS,RANK} combos) - are these two the same or different in the eyes of connect:
`array1 = [{1,0},{1,1},{1,2}]`
`array2 = [{1,2},{1,1},{1,0}]`